src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -98,10 +99,7 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                     }
                     catch (Exception ex)
                     {
-                        _logger.LogInformation(ex, "Failed to parse the query parameter. Skipping.");
-
-                        // There was a problem parsing the parameter. Add it to list of unsupported parameters.
-                        unsupportedSearchParameters.Add(query);
+                        throw new BadRequestException(ex.Message);
```

Comment:
Does this change the behavior in the case where we just don't support a search parameter? 

For example, if somebody were to send `Location?type1=test` would this return an error?

Reply:
Also, there's some discussion around this in the spec: https://www.hl7.org/fhir/search.html#errors

Reply:
I think it changes the behavior for only those requests which are syntactically incorrect (validation failed by HL7 package). "Location?type1=test" still returns result (with 200 OK) ignoring the unsupported params.

Reply:
Can we put comment here just to describe this briefly? I think it will help whoever reading the code.